src/emscripten-optimizer/simple_ast.h

```diff
@@ -920,22 +920,22 @@ struct JSPrinter {
         char *test = end;
         // remove zeros, and also doubles can use at most 24 digits, we can truncate any extras even if not zero
         while ((*test == '0' || test - buffer > 24) && test > buffer) test--;
-        int num = end - test;
+        size_t num = end - test;
         if (num >= 3) {
           test++;
           test[0] = 'e';
           if (num < 10) {
-            test[1] = '0' + num;
+            test[1] = (char)('0' + num);
```

Comment:
Right; `'0' + num` promotes to int because `num` is an int. Actually what I had in mind wouldn't really work. I think let's just put the casts on `num` expressions directly rather than the promoted result, e.g. `test[1] = '0' + (char)(num / 10);` That makes it clearer that there won't be overflow (since we are testing the value of `num`) and prevents the promotion altogether.